files/generate_system_metrics

```diff
@@ -31,6 +31,7 @@ module SystemMetrics
   # @attr [string] metrics_dir The puppet_metrics_collector output directory.
   # @attr [boolean] verbose Verbose output
   # @attr [string] hostname Name of the host the metrics are from. In directory name and json file.
+  # @attr [obj] time_stamp_obj Time object to use for generating the filename
```

Comment:
I don't understand why this is an instance variable.  This is set to `nil` during initialization and to different values within different methods.  Given the code in this PR, it seems like this should be treated as a local variable to each method.  Also, this particular line documents it as an `attr`, for which I would expect it to be created with [attr accessor](https://ruby-doc.org/core-2.5.0/Module.html#method-i-attr_accessor) and have read/write accessor methods.

Reply:
pinged on slack... I seem to have to choose between it being passed through nearly all the methods in the class, or making it a class attribute so that it is available to the few ones that actually need it.  I don't like either choice really.  But struggling to find something in the middle.

Reply:
As for the attr thing... I was following this.  https://kapeli.com/cheat_sheets/Yard.docset/Contents/Resources/Documents/index . I don't see another way to document it.  It's really just an internal variable...